app/src/main/java/com/github/se/gomeet/viewmodel/EventViewModel.kt

```diff
@@ -428,6 +428,22 @@ class EventViewModel(val currentUID: String? = null) : ViewModel() {
         event.copy(pendingParticipants = event.pendingParticipants.minus(userId)))
   }
 
+  /**
+   * Update the posts field of the event by removing the given post to the list.
+   *
+   * @param event the event to update
+   * @param post the post to remove from the event
+   */
+  fun deletePost(event: Event, post: Post) {
```

Comment:
Ideally this function would update only the `posts` list in the concerned event rather than the whole event (unnecessary network traffic), but again, as long as it works for now.

Reply:
I don't understand what you are saying. To update the list of posts I need to have access to the `Event` where this list is stored.

Reply:
It's possible to update only one field of a document in firebase, rather than updating the whole document. And in this situation, since you are modifying only the `posts` field of a document in the `events` Firebase folder, it would make more sense to reupload only the `posts` field rather than the whole event.

Reply:
Got it, will do it for next time, thanks.